common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1978,7 +1990,7 @@ def __init__(
         # explicit field_data during construct_xblock.
         kwargs.setdefault('id_reader', getattr(descriptor_runtime, 'id_reader', OpaqueKeyReader()))
         kwargs.setdefault('id_generator', getattr(descriptor_runtime, 'id_generator', AsideKeyGenerator()))
-        super().__init__(field_data=field_data, **kwargs)
+        super().__init__(**kwargs)
```

Comment:
We don't need to handle marking filed_data deprecated  as the Runtime class also considers this as a deprecated field and raises warning:
https://github.com/openedx/XBlock/blob/288f01476bc571664d6a8ed990528ca3341e6bb4/xblock/runtime.py#L577-L596

Reply:
Also, the Runtime handles assign deprecated field_data param to services [here](https://github.com/openedx/XBlock/blob/288f01476bc571664d6a8ed990528ca3341e6bb4/xblock/runtime.py#L555-L562).